hwtracer/src/decode/ykpt/mod.rs

```diff
@@ -38,32 +80,151 @@ impl TraceDecoder for YkPTTraceDecoder {
         let itr = YkPTBlockIterator {
             errored: false,
             parser: PacketParser::new(trace.bytes()),
-            cur_loc: ObjLoc::OtherObjOrUnknown,
+            cur_loc: ObjLoc::OtherObjOrUnknown(None),
             tnts: VecDeque::new(),
+            comprets: CompressedReturns::new(),
             pge: false,
         };
         Box::new(itr)
     }
 }
 
+/// The virtual address ranges of segments that we may need to disassemble.
+static CODE_SEGS: LazyLock<CodeSegs> = LazyLock::new(|| {
+    let mut segs = Vec::new();
+    for obj in phdrs::objects() {
+        let obj_base = obj.addr();
+        for hdr in obj.iter_phdrs() {
+            if (hdr.flags() & libc::PF_W) == 0 {
+                let vaddr = usize::try_from(obj_base + hdr.vaddr()).unwrap();
+                let memsz = usize::try_from(hdr.memsz()).unwrap();
+                let key = vaddr..(vaddr + memsz);
+                segs.push((key.clone(), ()));
+            }
+        }
+    }
+    let tree = segs.into_iter().collect::<IntervalTree<usize, ()>>();
+    CodeSegs { tree }
+});
+
+const PT_MAX_COMPRETS: usize = 64;
```

Comment:
Doc string?

Reply:
Fixed in d4bdb709ef7e773e42409937f0f7d1e6452ff43e

(Sorry, got mixed into another change, but should be easy to see)